### PR TITLE
Odin 20% Monday improvements

### PR DESCRIPTION
**Pull request by miguelbernadi**
These are the changes for the 20% day on Odin. The commits have been squashed logically to ease review one commit at a time, though the PR is very complex.

### CODE REVIEW

**Comment by ifosch** [comment]
+1

File: mcc/odin/odin/odin.go [hunk 1]
```diff
@@ -38,17 +39,15 @@ func (params CreateDBParams) GetRestoreDBInstanceFromDBSnapshotInput(
...
 	if params.OriginalInstanceName == "" {
-		log.Fatalf(
-			"Original instance is required",
-		)
+		return nil, fmt.Errorf("Original instance is required")
 	}
 	snapshot, err = GetLastSnapshot(params.OriginalInstanceName, svc)
 	if err != nil {
-		log.Fatalf(
+		return nil, fmt.Errorf(
```
Comment (dsager) [inline mcc/odin/odin/odin.go]: not sure about go, but in most languages multiple return values are considered bad practice...

File: mcc/odin/odin/odin_test.go [hunk 2]
```diff
@@ -545,6 +545,35 @@ var getRestoreDBInstanceFromDBSnapshotInputCases = []getRestoreDBInstanceFromDBS
...
+		createDBParams: CreateDBParams{
+			DBInstanceType:       "db.m1.medium",
+			DBUser:               "owner",
+			DBPassword:           "password",
+			Size:                 5,
+			OriginalInstanceName: "develop",
+		},
+		snapshot: exampleSnapshot1,
+		expectedRestoreDBInstanceFromDBSnapshotInput: nil,
+		expectedError:                                "Couldn't find snapshot for develop instance",
```
Comment (dsager) [inline mcc/odin/odin/odin_test.go]: the indentation is weird, in this case it's aligned, for the other values above not?

**Comment by dsager** [comment]
+1

File: mcc/odin/odin/odin.go (hunk 1 above)
Comment (miguelbernadi) [reply mcc/odin/odin/odin.go]: The function definition states you have to return 2 values. In Go, return order is always fixed (it's as declared in the function signature) and not handling all returned values causes a compiler error.

File: mcc/odin/odin/odin_test.go (hunk 2 above)
Comment (miguelbernadi) [reply mcc/odin/odin/odin_test.go]: Yes, indentation is kind of inconsistent, but it's created by the gofmt tool, so even if we manually tweak it it's reformatted automatically when the file is verified or processed by the linter. We would have to manually maintain a different formatting every time we change something in the file.

Comment (miguelbernadi) [reply mcc/odin/odin/odin_test.go]: will try to look into it. It would be either expected behaviour or a bug in the Go tooling.

**Pull request merged**